homeassistant/components/homekit/type_media_players.py

```diff
@@ -142,3 +170,211 @@ def update_state(self, new_state):
+                      FEATURE_VOLUME_STEP: None}
+        feature_list = self.config[CONF_FEATURE_LIST]
+
+        self.sources = []
+
+        self.category = CATEGORY_TELEVISION
+        television = self.add_preload_service(SERV_TELEVISION,
+                                              [
+                                                  CHAR_CONFIGURED_NAME,
+                                                  CHAR_REMOTE_KEY
+                                              ])
+        television.configure_char(CHAR_CONFIGURED_NAME,
+                                  value=self.display_name)
+        television.configure_char(CHAR_SLEEP_DISCOVER_MODE, value=True)
+        self.chars[FEATURE_ON_OFF] = television.configure_char(
+            CHAR_ACTIVE, setter_callback=self.set_on_off)
+
+        if FEATURE_PLAY_PAUSE in feature_list:
+            self.chars[FEATURE_PLAY_PAUSE] \
+                    = television.configure_char(
+                        CHAR_REMOTE_KEY,
+                        setter_callback=self.set_remote_key)
+
+        if FEATURE_TOGGLE_MUTE in feature_list:
+
+            television_speaker = self.add_preload_service(
+                SERV_TELEVISION_SPEAKER, [
+                    CHAR_NAME, CHAR_ACTIVE,
+                    CHAR_VOLUME_CONTROL_TYPE,
+                    CHAR_VOLUME_SELECTOR,
+                    CHAR_VOLUME
+                ])
+            television.add_linked_service(television_speaker)
+
+            name = '{} {}'.format(self.display_name, 'Volume')
+            television_speaker.configure_char(CHAR_NAME, value=name)
+            television_speaker.configure_char(CHAR_ACTIVE, value=1)
+
+            self.chars[FEATURE_TOGGLE_MUTE] \
+                = television_speaker.configure_char(
+                    CHAR_MUTE, value=False,
+                    setter_callback=self.set_toggle_mute)
+
+            if FEATURE_VOLUME_STEP in feature_list:
+                television_speaker.configure_char(CHAR_VOLUME_CONTROL_TYPE,
+                                                  value=1)
+                self.chars[FEATURE_VOLUME_STEP] = television_speaker.\
+                    configure_char(
+                        CHAR_VOLUME_SELECTOR,
+                        setter_callback=self.set_volume_step)
+
+        if FEATURE_SELECT_SOURCE in feature_list:
+
+            self.chars[FEATURE_SELECT_SOURCE] \
+                    = television.configure_char(
+                        CHAR_ACTIVE_IDENTIFIER,
+                        setter_callback=self.set_input_source)
+
+            self.sources = self.hass.states.get(
+                self.entity_id).attributes.get('source_list')
+            if self.sources:
+                for index, source in enumerate(self.sources):
+                    input_service = self.add_preload_service(
+                        SERV_INPUT_SOURCE, [
+                            CHAR_IDENTIFIER,
+                            CHAR_NAME
+                        ])
```

Comment:
Just one line for the four above if that fits: `SERV_INPUT_SOURCE, [CHAR_.., CHAR_NAME])`